modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateInput.es.js

```diff
@@ -9,31 +9,73 @@ class DateInput extends React.Component {
 		value: propTypes.string
 	};
 
+	constructor(props) {
+		super(props);
+
+		const date = new Date(props.value);
+
+		this.state = {
+			value: jsDatetoYYYYMMDD(date)
+		};
+	}
+
+	shouldComponentUpdate(nextProps) {
```

Comment:
You should probably use `getDerivedStateFromProps` for this lifecycle. https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
Thanks!! This is exactly what we need.